doc/dynamic-environments/configuration.mkd

```diff
@@ -454,3 +477,46 @@ This will create the following directory structure:
 |-- app1_production  # app1 data repository, production branch
 |-- app1_develop     # app1 data repository, develop branch
 ```
+
+### Filtering branches
+
+#### Filtering branches - Exclude branches with a prefix
+
+If you would like to filter out all branches starting with and including "dev-".
+
+```yaml
+---
+sources:
+  app1_data:
+    remote: 'git://git-server.site/my-org/app1-hieradata'
+    basedir: '/etc/puppet/hieradata'
+    branch_filter: '^(?!dev-).*$'
+```
+
+#### Filter branches - Include branches with a prefix
+
+If you would like to only deploy branches starting with a prefix "prod-" followed by word characters. Note: this does NOT match the "prod-" branch exactly due to the \w+.
+
+```yaml
+---
+sources:
+  app1_data:
+    remote: 'git://git-server.site/my-org/app1-hieradata'
+    basedir: '/etc/puppet/hieradata'
+    branch_filter: '^(prod-)\w+$'
```

Comment:
You shouldn't need a grouping here at all?

Reply:
No, it doesn't matter in terms of functionality, but does help to logically separate out the prefix from the "extra stuff at the end".